aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsEndpointDecorator.kt

```diff
@@ -65,143 +77,144 @@ class AwsEndpointDecorator : RustCodegenDecorator<ClientProtocolGenerator, Clien
+                    self.endpoint_resolver = Some(std::sync::Arc::new(#{EndpointShim}::from_resolver(endpoint_resolver)) as _);
+                    self
+                }
 
-                    /// Sets the endpoint resolver to use when making requests.
-                    pub fn set_endpoint_resolver(&mut self, endpoint_resolver: Option<std::sync::Arc<dyn #{ResolveAwsEndpoint}>>) -> &mut Self {
-                        self.endpoint_resolver = endpoint_resolver.map(|res|std::sync::Arc::new(#{EndpointShim}::from_arc(res) ) as _);
-                        self
-                    }
-                    """,
-                    *codegenScope,
-                )
+                ##[deprecated(note = "use endpoint_url or set the endpoint resolver directly")]
+                /// Sets the endpoint resolver to use when making requests.
+                ///
+                /// This method is deprecated, use [`Builder::endpoint_url`] or [`Builder::endpoint_resolver`] instead.
+                pub fn set_aws_endpoint_resolver(&mut self, endpoint_resolver: Option<std::sync::Arc<dyn #{ResolveAwsEndpoint}>>) -> &mut Self {
+                    self.endpoint_resolver = endpoint_resolver.map(|res|std::sync::Arc::new(#{EndpointShim}::from_arc(res) ) as _);
+                    self
+                }
 
-            ServiceConfig.BuilderBuild -> {
-                val resolverGenerator = EndpointResolverGenerator(codegenContext, endpointData)
-                rustTemplate(
-                    """
-                    endpoint_resolver: self.endpoint_resolver.unwrap_or_else(||
-                        std::sync::Arc::new(#{EndpointShim}::from_resolver(#{Resolver}()))
-                    ),
-                    """,
-                    *codegenScope, "Resolver" to resolverGenerator.resolver(),
-                )
-            }
+                /// Sets the endpoint url used to communicate with this service
+                ///
+                /// Note: this is used in combination with other endpoint rules, e.g. an API that applies a host-label prefix
+                /// will be prefixed onto this URL. To fully override the endpoint resolver, use
+                /// [`Builder::endpoint_resolver`].
+                pub fn endpoint_url(mut self, endpoint_url: impl Into<String>) -> Self {
+                    self.endpoint_url = Some(endpoint_url.into());
+                    self
+                }
 
-            else -> emptySection
-        }
-    }
-}
+                /// Sets the endpoint url used to communicate with this service
+                ///
+                /// Note: this is used in combination with other endpoint rules, e.g. an API that applies a host-label prefix
+                /// will be prefixed onto this URL. To fully override the endpoint resolver, use
+                /// [`Builder::endpoint_resolver`].
+                pub fn set_endpoint_url(&mut self, endpoint_url: Option<String>) -> &mut Self {
```

Comment:
Is the reason for not taking a URL here that we don't want to expose a 3rd party type?

Reply:
Just nicer for users since we can take static strings and we're going to convert it back to a string anyway to pass it into ep2